azext_edge/edge/providers/orchestration/connected_cluster.py

```diff
@@ -48,14 +105,38 @@ def extensions(self) -> List[dict]:
         return additional_properties.get("value", [])
 
     def get_custom_location_for_namespace(self, namespace: str) -> Optional[dict]:
-        query = f"""
-        resources
-        | where type =~ 'microsoft.extendedlocation/customlocations'
-        | where properties.hostResourceId =~ '{self.resource_id}'
-        | where properties.namespace =~ '{namespace}'
-        | project id, name, location, properties, apiVersion
-        """
+        query = QUERIES["get_custom_location_for_namespace"].format(resource_id=self.resource_id, namespace=namespace)
+
+        result = self.resource_graph.query_resources(query=query)
+        return self._process_query_result(result, first=True)
+
+    def get_aio_extensions(self) -> Optional[List[dict]]:
+        query = QUERIES["get_aio_extensions"].format(resource_id=self.resource_id)
+        # TODO - @digimaun microsoft.azurekeyvaultsecretsprovider optionality
+
+        result = self.resource_graph.query_resources(query=query)
+        return self._process_query_result(result)
+
+    def get_aio_custom_locations(self) -> Optional[List[dict]]:
+        query = QUERIES["get_aio_custom_locations"].format(resource_id=self.resource_id)
 
         result = self.resource_graph.query_resources(query=query)
+        return self._process_query_result(result)
+
+    def get_aio_resources(self, custom_location_id: str) -> Optional[List[dict]]:
+        query = QUERIES["get_aio_resources"].format(custom_location_id=custom_location_id)
+
+        result = self.resource_graph.query_resources(query=query)
+        return self._process_query_result(result)
+
+    def get_resource_sync_rules(self, custom_location_id: str) -> Optional[List[dict]]:
+        query = QUERIES["get_resource_sync_rules"].format(custom_location_id=custom_location_id)
+
+        result = self.resource_graph.query_resources(query=query)
+        return self._process_query_result(result)
+
+    def _process_query_result(self, result: dict, first: bool = False) -> Optional[Union[dict, List[dict]]]:
```

Comment:
why not just include first in the query?

Reply:
The ARG query? Like limit 1? In this case it would still return an array.

Reply:
hmm we could just move the get one logic to where it is used (get_custom_location_for_namespace) to minimize params here

Reply:
also limit 1 would reduce the amount of data we receive (even tho we don't sift through it here)

Reply:
> hmm we could just move the get one logic to where it is used (get_custom_location_for_namespace) to minimize params here

The problem with this is having to duplicate logic in `get_custom_location_for_namespace`, to check if the returned value is truthy or not which already happens in `_process_query_result`.

Also when you say minimize parameters, there are only 2 parameters `_process_query_result`.

> also limit 1 would reduce the amount of data we receive (even tho we don't sift through it here)

You asked a related question here, where I am trying to convey there is at most 1 custom location associated to a namespace, so adding limit 1 makes a more verbose query for what value add?

![image](https://github.com/Azure/azure-iot-ops-cli-extension/assets/28658112/d7774cc9-1cd8-4464-9179-72093d02af0f)

Reply:
regarding the first one:
yea - I know it is only two params,  wanted to keep it as clean as possible for now. If there will be dup code otherwise, then lets just keep the params

regarding the second one:
lets leave it be then (without the limit 1)